ADAL/src/ADAuthenticationContext.m

```diff
@@ -118,10 +131,12 @@ - (id)initWithAuthority:(NSString *)authority
     ADTokenCache* cache = [ADTokenCache new];
     [cache setDelegate:delegate];
     
-    return [self initWithAuthority:authority
+    self = [self initWithAuthority:authority
                  validateAuthority:validateAuthority
                         tokenCache:cache
                              error:error];
+    SAFE_ARC_RELEASE(cache);
+    return self;
```

Comment:
Once we pass it into the authentication context it takes ownership of it and we don't need it anymore so we release it.